src/middlewared/middlewared/plugins/truenas_connect/update.py

```diff
@@ -64,18 +69,53 @@ async def config_extend(self, config):
     @private
     async def validate_data(self, old_config, data):
         verrors = ValidationErrors()
-        if data['enabled'] and not data['ips']:
-            verrors.add('tn_connect_update.ips', 'This field is required when TrueNAS Connect is enabled')
+
+        # Ensure at least one interface or at least one IP is provided
+        if data['enabled'] and not data['ips'] and not data['interfaces']:
+            verrors.add(
+                'tn_connect_update',
+                'At least one IP or interface must be provided when TrueNAS Connect is enabled'
+            )
 
         data['ips'] = [str(ip) for ip in data['ips']]
 
+        # Validate interfaces exist
+        if data['interfaces']:
+            all_interfaces = await self.middleware.call('interface.query')
```

Comment:
No reason to call `interface.query` here when all you're doing is checking the names.
```
'interface.query', [], {'extra': {'retrieve_names_only': True}}
```